### PR TITLE
Demo of Fully Static Nuxt Site

### PR DESCRIPTION
**Pull request by TeddTech**
* [x] This pull request has been linted by running `npm run lint` without errors
* [x] This pull request has been tested by running `npm run dev` and reviewing affected routes
* [x] This pull request has been built by running `npm run generate` without errors
* [x] This isn't a duplicate of an existing pull request

## Description

Demo of fully static nuxt site

## Steps to test

1. run `npm run generate`
2. run `npx serve dist`
3. open up network tab in inspect and click around
4. monitor network tab

**Expected behavior:** 

Only making api calls during search

### CODE REVIEW

File: nuxt.config.js [hunk 1]
```diff
@@ -44,6 +46,29 @@ export default {
 			{ rel: "icon", type: "image/png", href: "/favicon.png" }
 		]
 	},
+	render: { resourceHints: false },
+	hooks: {
+		// remove uneeded js scripts
+		"generate:page": (page) => {
+			const doc = cheerio.load(page.html);
+			doc("body script[src='/_nuxt/aac6f7dcb0388507b165.js']").remove();
```
Comment (cindyli) [inline nuxt.config.js:54]: Removing javascript on static pages using hardcoded nuxt generated names is not reliable as the script names could change from build to build. I did get a different set of names on my computer.

I wonder if you really need this `hooks` block to remove javascript from static pages? When testing this pull request, I noticed the fetch of the data comes from the local payloads that are saved in the `dist` directory with your addition of `payload` field in the nuxt.config.js `generate` block, which should be enough to deal with the remote fetching problem.

File: pages/_slug/index.vue [hunk 2]
```diff
@@ -37,8 +26,12 @@ export default {
...
 		};
 	},
-	fetch ({ store }) {
-		return store.dispatch("fetchSitePages");
+	asyncData ({ $axios, $payloadURL, route, store }) {
+		if (process.static && process.client && $payloadURL) {
+			return $axios.$get($payloadURL(route));
+		}
+
+		return store.state.sitePages.find(oneSitePage => oneSitePage.slug === route.params.slug);
```
Comment (cindyli) [inline pages/_slug/index.vue:34]: Shall  `await store.dispatch("fetchSitePages");` be added before this line?

File: pages/index.vue [hunk 3]
```diff
@@ -41,8 +35,12 @@ export default {
...
 		};
 	},
-	fetch ({ store }) {
-		return store.dispatch("fetchSitePages");
+	asyncData ({ $axios, $payloadURL, route, store }) {
+		if (process.static && process.client && $payloadURL) {
+			return $axios.$get($payloadURL(route));
+		}
+
+		return store.state.sitePages.find(oneSitePage => oneSitePage.slug === "home");
```
Comment (cindyli) [inline pages/index.vue:43]: Same as above: add `await store.dispatch("fetchSitePages");` before this line?

File: pages/_slug/index.vue (hunk 2 above)
Comment (TeddTech) [reply pages/_slug/index.vue:34]: No need because it is called in `actions.js` within `nuxtServerInit`

File: pages/index.vue (hunk 3 above)
Comment (TeddTech) [reply pages/index.vue:43]: No need because it is called in `actions.js` within `nuxtServerInit`

File: nuxt.config.js (hunk 1 above)
Comment (TeddTech) [reply nuxt.config.js:54]: Yes, you are absolutely right. It is enough to deal with the remote fetching problem.

I included the removal of js scripts to show what is possible and different directions we can go. I will expand on this further in the check-in today.

File: pages/index.vue (hunk 3 above)
Comment (cindyli) [reply pages/index.vue:43]: Makes sense. Thanks.

**Pull request closed without merging**